src/GovUk.Education.ExploreEducationStatistics.Content.Services/MethodologyService.cs

```diff
@@ -125,5 +131,82 @@ private async Task<List<MethodologyVersionSummaryViewModel>> BuildMethodologiesF
+        {
+            var slugChanged = originalSlug != updatedSlug;
+
+            var ownedMethodology = await _contentDbContext
+                .PublicationMethodologies
+                .Include(pm => pm.Methodology.LatestPublishedVersion)
+                .Include(pm => pm.Methodology.Versions)
+                .Where(pm => pm.PublicationId == publicationId && pm.Owner)
+                .Select(pm => pm.Methodology)
+                .SingleOrDefaultAsync();
+
+            if (ownedMethodology == null)
+            {
+                return;
+            }
+
+            ownedMethodology.OwningPublicationTitle = updatedTitle;
+            ownedMethodology.OwningPublicationSlug = updatedSlug;
+
+            _contentDbContext.Methodologies.Update(ownedMethodology);
+
+            if (slugChanged)
+            {
+                // A redirect is only needed for the LatestPublishedVersion.
+                // Unpublished methodology's don't need a redirect - they're not live.
+                // An unpublished amendment doesn't need a redirect because:
+                // - if it uses OwningPublicationSlug, it is covered by the LatestPublishedVersion redirect created here
+                // - if it uses AlternativeSlug, a redirect would have been created at the time the AlternativeSlug
+                //   was set (and that redirect will become active when that version is published).
+                if (ownedMethodology.LatestPublishedVersion is { AlternativeSlug: null }
+                    // guard against duplicates due to users making multiple publication slug changes
+                    && !await _contentDbContext.MethodologyRedirects
+                        .AnyAsync(mr =>
+                            mr.MethodologyVersionId == ownedMethodology.LatestPublishedVersionId
+                            && mr.Slug == originalSlug))
+                {
+                    var redirect = new MethodologyRedirect
+                    {
+                        MethodologyVersion = ownedMethodology.LatestPublishedVersion,
+                        Slug = originalSlug,
+                    };
+                    _contentDbContext.MethodologyRedirects.Add(redirect);
+
+                    // It's possible we now have two redirects from the same slug. This happens if:
+                    // - An unpublished amendment sets an AlternativeSlug
+                    // - Then the OwningPublicationSlug changes when the LatestPublishedVersion is
+                    //   inheriting it.
+                    // We must delete the unpublished amendment's redirect, as otherwise if the unpublished version
+                    // changes its slug again, no redirect will be created for the methodology's new slug. (If updating
+                    // an unpublished version's slug multiple times, a redirect is only created the first time - see
+                    // MethodologyUpdate code). If this doesn't make any sense, see the associated unit test
```

Comment:
Instead of `If this doesn't make any sense, see the associated unit test`, I would instead write `See MethodologyServiceTests.PublicationTitleOrSlugChanged_NastyScenario() for an example.`